### PR TITLE
cli: added legacy and stream flags, changed prompt to reflect usage

### PR DESCRIPTION
**Pull request by abdullahzameek**
Hi @nicks 

Would this approach count as an "intermediary" solution to the Epic 7733?

### CODE REVIEW

**Comment by abdullahzameek** [comment]
I had a look at the tests at failed in `build-windows` and I'm kinda confused


`=== Failed
=== FAIL: internal/tiltfile TestKustomize (0.27s)
    tiltfile_test.go:5319: Traceback (most recent call last):
          C:\Users\circleci\AppData\Local\Temp\TestKustomize181797138\Tiltfile:4:19: in <toplevel>
          <builtin>: in kustomize
        Error: command ["kustomize" "build" "C:\\Users\\circleci\\AppData\\Local\\Temp\\TestKustomize181797138"] failed.
        error: exit status 1
        stdout: ""
        stderr: "Error: unable to find one of 'kustomization.yaml', 'kustomization.yml' or 'Kustomization' in directory 'C:\\Users\\circleci\\AppData\\Local\\Temp\\kustomize-334198219\\repo'\n"

=== FAIL: internal/tiltfile TestKustomization (0.24s)
    tiltfile_test.go:5319: Traceback (most recent call last):
          C:\Users\circleci\AppData\Local\Temp\TestKustomization222459412\Tiltfile:4:19: in <toplevel>
          <builtin>: in kustomize
        Error: command ["kustomize" "build" "C:\\Users\\circleci\\AppData\\Local\\Temp\\TestKustomization222459412"] failed.
        error: exit status 1
        stdout: ""
        stderr: "Error: unable to find one of 'kustomization.yaml', 'kustomization.yml' or 'Kustomization' in directory 'C:\\Users\\circleci\\AppData\\Local\\Temp\\kustomize-818785899\\repo'\n"
`

File: internal/cli/up.go [hunk 1]
```diff
@@ -84,6 +87,8 @@ local resources--i.e. those using serve_cmd--are terminated when you exit Tilt.
 		fmt.Sprintf("Control the strategy Tilt uses for updating instances. Possible values: %v", buildcontrol.AllUpdateModes))
 	cmd.Flags().StringVar(&c.traceTags, "traceTags", "", "tags to add to spans for easy querying, of the form: key1=val1,key2=val2")
 	cmd.Flags().BoolVar(&c.hud, "hud", true, "If true, tilt will open in HUD mode.")
+	cmd.Flags().BoolVar(&c.legacy, "legacy", true, "If true, tilt will open in legacy terminal mode.")
```
Comment (nicks) [inline internal/cli/up.go:90]: this should default to false (the third argument)

File: internal/cli/up.go [hunk 2]
```diff
@@ -84,6 +87,8 @@ local resources--i.e. those using serve_cmd--are terminated when you exit Tilt.
...
+	cmd.Flags().BoolVar(&c.stream, "stream", true, "If true, tilt will stream logs in the terminal.")
```
Comment (nicks) [inline internal/cli/up.go:91]: this should also default to false

File: internal/cli/up.go [hunk 3]
```diff
@@ -95,6 +100,8 @@ local resources--i.e. those using serve_cmd--are terminated when you exit Tilt.
 	cmd.PreRun = func(cmd *cobra.Command, args []string) {
 		c.hudFlagExplicitlySet = cmd.Flag("hud").Changed
 		c.watchFlagExplicitlySet = cmd.Flag("watch").Changed
+		c.legacyFlagExplicitlySet = cmd.Flag("legacy").Changed
```
Comment (nicks) [inline internal/cli/up.go:103]: i understand why you copied this from the above flags. but this pattern should really only be used for new flags where we're trying to preserve some backwards-compatibility. it never makes sense to read `Changed` on newly introduced flags

File: internal/hud/prompt/prompt.go [hunk 4]
```diff
@@ -208,9 +208,9 @@ func (p *TerminalPrompt) OnChange(ctx context.Context, st store.RStore) {
 					st.Dispatch(SwitchTerminalModeAction{Mode: store.TerminalModeStream})
 					msg.stopCh <- true
 
-				case 'h':
-					warningMsg := "We're removing the HUD key shortcut from the prompt.\n" +
-						"In future releases, open the HUD with: tilt up --hud=true"
+				case 't':
```
Comment (nicks) [inline internal/hud/prompt/prompt.go:211]: can we also continue to support 'h'?

File: internal/cli/up.go (hunk 3 above)
Comment (abdullahzameek) [reply internal/cli/up.go:103]: @nicks I'm taking a second look at this, and I'm slightly confused.
Wouldn't I need this boolean to see if the user has explicitly invoked the --legacy or --stream flags?

**Comment by abdullahzameek** [comment]
@nicks Made changes as requested!

File: internal/hud/prompt/prompt.go (hunk 4 above)
Comment (nicks) [reply internal/hud/prompt/prompt.go:211]: you can do this as

case 't', 'h':

https://gobyexample.com/switch

File: internal/hud/prompt/prompt.go [hunk 5]
```diff
@@ -208,9 +208,22 @@ func (p *TerminalPrompt) OnChange(ctx context.Context, st store.RStore) {
 					st.Dispatch(SwitchTerminalModeAction{Mode: store.TerminalModeStream})
 					msg.stopCh <- true
 
+				case 't':
+					warningMsg := "We're removing the legacy terminal mode key shortcut from the prompt.\n" +
```
Comment (nicks) [inline internal/hud/prompt/prompt.go:212]: let's remove the warning message and the Sleep too, since this isn't going away

Comment (abdullahzameek) [reply internal/hud/prompt/prompt.go:212]: Oooh another neat Go syntax trick! Will make those changes!

**Pull request merged**